openpype/hosts/batchpublisher/controller.py

```diff
@@ -0,0 +1,219 @@
+import glob
+import os
+
+from openpype.settings import get_project_settings
+from openpype.client.entities import (
+    get_projects,
+    get_assets,
+)
+from openpype.hosts.batchpublisher import publish
+
+
+# TODO: add to OpenPype settings so other studios can change
+FILE_MAPPINGS = [
+    {
+        "glob": "*/fbx/*.fbx",
+        "is_sequence": False,
+        "product_type": "model",
+    }
+]
+
+
+class IngestFile(object):
+    def __init__(
+        self,
+        filepath,
+        product_type,
+        product_name,
+        representation_name,
+        version=None,
+        enabled=True,
+        folder_path=None,
+        task_name=None
+    ):
+        self.enabled = enabled
+        self.filepath = filepath
+        self.product_type = product_type
+        self.product_name = product_name or ""
+        self.representation_name = representation_name
+        self.version = version
+        self.folder_path = folder_path or ""
+        self.task_name = task_name or ""
+        self.task_names = []
+
+    @property
+    def defined(self):
+        return all([
+            bool(self.filepath),
+            bool(self.folder_path),
+            bool(self.task_name),
+            bool(self.product_type),
+            bool(self.product_name),
+            bool(self.representation_name)])
```

Comment:
The `bool` here is redundant because `all` would already check the "if" state of the values.

Reply:
yeah true I corrected this.

I just liked to make it obvious its checking the bool condition of each